org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/NodeModelBuilder.java

```diff
@@ -273,16 +273,17 @@ protected void replaceChildren(AbstractNode oldNode, AbstractNode newNode) {
 		}
 	}
 	
-	public void replaceAndTransferLookAhead(INode oldNode, INode newRootNode) {
+	private AbstractNode doReplaceAndTransferLookAhead(INode oldNode, INode newRootNode) {
 		AbstractNode newNode = ((CompositeNode) newRootNode).basicGetFirstChild();
 		replaceWithoutChildren((AbstractNode) oldNode, newNode);
 		if (oldNode instanceof ICompositeNode && newNode instanceof CompositeNode) {
 			CompositeNode newCompositeNode = (CompositeNode) newNode;
 			newCompositeNode.basicSetLookAhead(((ICompositeNode) oldNode).getLookAhead());
 		}
-		ICompositeNode root = newNode.getRootNode();
-		BidiTreeIterator<AbstractNode> iterator = ((AbstractNode) root).basicIterator();
-		int offset = 0;
+		return newNode;
+	}
+
+	private void adjustLookAhead(int offset, BidiTreeIterator<AbstractNode> iterator) {
```

Comment:
This method sets the total offsets into the leaf nodes. It's not related to look ahead

Reply:
Thanks for your patience. It only highlights my bad understanding of the code.

Lucky for me, I am only submitting working code that someone else did many years ago, I only need to extract the changes  in a way that they are mergeable in sensible and usable by others in chucks to xtext-core :)